nexus/tests/integration_tests/console_api.rs

```diff
@@ -206,6 +209,40 @@ async fn test_absolute_static_dir() {
     cptestctx.teardown().await;
 }
 
+#[nexus_test]
+async fn test_session_me(cptestctx: &ControlPlaneTestContext) {
+    let testctx = &cptestctx.external_client;
+
+    // hitting /session/me without being logged in is a 401
+    RequestBuilder::new(&testctx, Method::GET, "/session/me")
+        .expect_status(Some(StatusCode::UNAUTHORIZED))
+        .execute()
+        .await
+        .expect("failed to 401 on unauthed request");
+
+    // now make same request with auth
+    let priv_user = NexusRequest::object_get(testctx, "/session/me")
+        .authn_as(AuthnMode::PrivilegedUser)
+        .execute()
+        .await
+        .expect("failed to get current user")
+        .parsed_body::<views::User>()
+        .unwrap();
+
+    assert_eq!(priv_user.identity.name.as_str(), "test-privileged");
+
+    // same request as unprivileged user 403s because unprivileged user
```

Comment:
FWIW I don't think this is important to test.  i.e. arguably _any_ user should be able to see their own details -- if that doesn't work, that seems like a (perfectly fine) accident of our implementation, not designed behavior, right?

Reply:
Yeah, I think that's right. This came from me trying it with the unprivileged user assuming it would work, then finding it didn't work, and changing the assert to match the reality. But I agree, it doesn't matter.

Regardless, once I implement your suggestion to only pull user data that's already in auth context rather than making a separate database call, this case disappears.